res/ai/DefaultAIPlayer/TaskMovieDistributor.lua

```diff
@@ -222,64 +201,66 @@ function JobBuyStartProgramme:Tick()
 	local goodMovies = {}
 	-- sort lowest first
 	local sortByPrice = function(a, b)
-		return a:GetPrice(TVT.ME) < b:GetPrice(TVT.ME)
+		return a:GetPrice(TVT.ME)/a:GetBlocks() < b:GetPrice(TVT.ME)/b:GetBlocks()
 	end
 
 	-- add "okay" movies to the list of candidates
 	for k,v in pairs(movies) do
-		--avoid the absolute trash :-)
-		if (v:GetQuality() >= 0.10 and v:GetPrice(TVT.ME) <= startMovieBudgetMax) then
+		if v:GetQuality() < 0.10 or v:GetQualityRaw() < 0.25 then
+			--avoid the absolute trash :-)
+			self:LogDebug("IGNORING PROGRAMME (quality) "..v:getTitle())
+		elseif (v:isPaid() > 0 or v:getTopicality() < 0.25 or v:GetGenre() == TVT.Constants.ProgrammeGenre.Horror) then
 			--prevent other problematic start programmes: call-in, horror, too old
-			if (v:isPaid() > 0 or v:getTopicality() < 0.4 or v:GetGenre() == TVT.Constants.ProgrammeGenre.Horror) then
-				self:LogTrace("IGNORING PROGRAMME "..v:getTitle())
-			else
-				table.insert(goodMovies, v)
-			end
+			self:LogDebug("IGNORING PROGRAMME (old, genre) "..v:getTitle())
+		elseif v.isSeries() > 0 then
+			self:LogDebug("IGNORING PROGRAMME (series) "..v:getTitle())
+		elseif v:GetPrice(TVT.ME)/v:GetBlocks() > 50000 then 
+			self:LogDebug("IGNORING PROGRAMME (price) "..v:getTitle() .. " ".. v:GetPrice(TVT.ME)/v:GetBlocks())
+		elseif math.random(0,10) > 6 then
+			-- ignore randomly
+			self:LogDebug("IGNORING PROGRAMME (random) "..v:getTitle())
+		else
+			table.insert(goodMovies, v)
 		end
 	end
 	table.sort(goodMovies, sortByPrice)
 
 
 	local buyStartMovies = {}
 	for k,v in pairs(goodMovies) do
+		local price = v:GetPrice(TVT.ME)
 		-- stop iteration if getting low on budget
-		if startMoviesBudget < v:GetPrice(TVT.ME) then break end
-		-- a single licence could be more expensive than the average budget
-		if v:GetPrice(TVT.ME) <= startMovieBudgetMax then
-			table.insert(buyStartMovies, v)
-			startMoviesBudget = startMoviesBudget - v:GetPrice(TVT.ME)
-		end
+		if budget < price then break end
+		table.insert(buyStartMovies, v)
+		budget = budget - price
 	end
 
+	local boughtCount = 0
 	for k,v in pairs(buyStartMovies) do
-		--only buy whole start programme set if possible with budget
-		--else each one should be cheaper than the single licence limit
-		if (table.count(buyStartMovies) >= moviesNeeded or v:GetPrice(TVT.ME) < startMovieBudget) then
-			--self:LogDebug("Buying start programme licence: " .. v:GetTitle() .. " (id=" .. v:GetId() .. ", price=" .. v:GetPrice(TVT.ME) .. ", quality=" .. v:GetQuality())
-			self:LogInfo("Buying start programme licence: " .. v:GetTitle() .. " price=" .. v:GetPrice(TVT.ME))
-			TVT.md_doBuyProgrammeLicence(v:GetId())
-
-			--attention: we subtract from the overall "buying programme"
-			--           budget!
-			self.Task:PayFromBudget(v:GetPrice(TVT.ME))
-			self.Task.CurrentBargainBudget = self.Task.CurrentBargainBudget - v:GetPrice(TVT.ME)
-		end
+		--Fairness - leave licences for other players
```

Comment:
Hmm ... wirkt ein wenig "versteckt" - und ist ja an sich nur so lange relevant, bis es (weit) mehr DB-Eintraege gibt. Oder?

Reply:
Auch hier ergänze ich ein TODO. Eine weitere Alternative wäre, dass in den ersten Stunden des Spiels die Lizenzen häufiger (1x pro Stunde?) ausgetauscht werden. Dann könnte auch bei der aktuellen DB-Lage genügend Material dabei sein, dass die KI ihr Startprogramm zusammenbekommt.